jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/FloatVectorAccessor.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.salesforce.datacloud.jdbc.core.accessor.impl;
+
+import com.salesforce.datacloud.jdbc.core.accessor.QueryJDBCAccessor;
+import com.salesforce.datacloud.jdbc.core.accessor.QueryJDBCAccessorFactory;
+import com.salesforce.datacloud.jdbc.exception.DataCloudJDBCException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.sql.SQLException;
+import java.util.function.IntSupplier;
+import lombok.val;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.holders.NullableFloat4Holder;
+
+public class FloatVectorAccessor extends QueryJDBCAccessor {
+
+    private final Float4Vector vector;
+    private final NullableFloat4Holder holder;
+
+    private static final String INVALID_VALUE_ERROR_RESPONSE = "BigDecimal doesn't support Infinite/NaN";
```

Comment:
Should this be "float" not "BigDecimal"?

Reply:
this error message is basically when you call getBigDecimal(), so this should be BigDecimal. 🤔

Reply:
Got it 👍